### PR TITLE
fix(input): standardized check and radio alignment

### LINKED ISSUE #5802: Bug - Checkbox and Radio - Vertical misalignment
**Describe the problem**
Checkboxes and radio buttons are not properly vertically aligned. This even looks differently broken in firefox and chromium.

We see this in Cockpit, they've  been broken for a while throughout the 5.0.0-alpha and 5.0.0-prerelease series, and it still affects 5.0.0 final.

**How do you reproduce the problem?**

The rendering on https://www.patternfly.org/components/forms/checkbox itself looks good. Click on the codesandbox examples for [Controlled](https://codesandbox.io/s/frwy6w?file=/index.tsx) or [Uncontrolled](https://codesandbox.io/s/psj3tk) and you see the issue right away.

With Firefox 115 the labels are too high:

image

With Chromium 115 the labels are slightly too low (but almost good):

chromium

(Sorry for the frame noise, screenshotting elements in chromium is utterly broken..)

The effect is somehow more pronounced in Cockpit, where the Chromium screenshot looks like this:

chromium

Firefox looks similarly misaligned on codesandbox and Cockpit.

**Expected behavior**

Checkboxes and radio buttons are centrally vertically aligned in all browsers.

**Is this issue blocking you?**

No, we have a workaround.

**What is your environment?**
 - OS: Fedora 38
 - Browser: Firefox 115, Chromium 115, @garrett also tested with webkit
 - Version: PatternFly 5.0.0 final

**What is your product and what release date are you targeting?**

https://github.com/cockpit-project/cockpit ; we release continuously, but try to keep PF up to date.

**Any other information?**

@garrett has found a [PF CSS patch](https://github.com/cockpit-project/cockpit/commit/9f84d9ac98238b10ce9e17fdfb1737c116113c4e) that works in all three major browser engines, see https://github.com/cockpit-project/cockpit/pull/19138#issuecomment-1658327885 for detailed results. That also fixes the curious issue that checkboxes have a different height in different environments. With that, checkboxes and radios are good enough, and we will land the PF update to 5.0 final with this patch applied.

### PR DESCRIPTION
**Pull request by mattnolting**
closes #5802 

**Quicklinks:**
Checkbox: https://patternfly-pr-5947.surge.sh/components/forms/checkbox/
Radio: https://patternfly-pr-5947.surge.sh/components/forms/radio

**Still in flight:** 
[Table checkbox alignment ](https://patternfly-pr-5947.surge.sh/components/table#checkboxes-and-actions-example)

### CODE REVIEW

**Review by mcoker — COMMENTED** [review]
Check and radio examples look spot on! Great job with that. Looks like just the standalone variation is left.

Were you planning on adding the form element toolbar item in this PR or is that just for testing?

**Comment by mattnolting** [comment]
> Check and radio examples look spot on! Great job with that. Looks like just the standalone variation is left.

@mcoker standalone updated

> 
> Were you planning on adding the form element toolbar item in this PR or is that just for testing?

@mcoker I think it best to include form element toolbar item in this PR, but I can also put in a followup

**Comment by srambach** [comment]
Looking good. I think the formula is solid.

File: src/patternfly/components/Check/check.scss [hunk 1]
```diff
@@ -20,17 +21,26 @@
...
 }
 
+.#{$check}__input {
+  align-self: start;
+
+  // stylelint-disable property-no-vendor-prefix
+  -moz-transform: translateY(var(--#{$check}__input--TranslateY--moz)); // remove at breaking change - no longer needed
+  // stylelint-enable
+
+ // find height of single label, divide by two, offset by 50% of own height
```
Comment (mcoker) [inline src/patternfly/components/Check/check.scss:39]:
```suggestion
  // find height of single label, divide by two, offset by 50% of own height
```

File: src/patternfly/components/Check/check.scss [hunk 2]
```diff
@@ -20,17 +21,26 @@
   display: grid;
   grid-template-columns: auto 1fr;
   grid-gap: var(--#{$check}--GridGap);
+  align-items: baseline;
 
   &.pf-m-standalone {
     --#{$check}--GridGap: 0;
-    --#{$check}__input--TranslateY--moz: 0;
-
-    display: inline-grid;
```
Comment (mcoker) [inline src/patternfly/components/Check/check.scss:28]: Removing this will break being able to put things before/after it inline

**Review by mcoker — CHANGES_REQUESTED** [review]
L🎃TM except removing the inline display of the standalone variation, which I think we need - at least for the time being so that it isn't breaking.

**Review by mcoker — APPROVED** [review]
LGTM

File: src/patternfly/components/Toolbar/examples/Toolbar.md [hunk 3]
```diff
@@ -656,6 +656,8 @@ As the toolbar component is a hybrid layout and component, some of its elements
 | `.pf-m-full-height` | `.pf-v5-c-toolbar`, `.pf-v5-c-toolbar__content-section`, `.pf-v5-c-toolbar__group` | Modifies toolbar component to full height of its container and removes vertical padding. |
 | `.pf-m-static` | `.pf-v5-c-toolbar` | Modifies expandable content section to position itself to the nearest absolutely positioned parent outside of the toolbar component. This is used primarily for masthead toolbar. |
 | `.pf-m-expanded` | `.pf-v5-c-toolbar__expandable-content` | Modifies expandable content section for the expanded state. |
+| `.pf-m-label` | `.pf-v5-c-toolbar__item` | Initiates label vertical positioning. |
+| `.pf-m-form-element` | `.pf-v5-c-toolbar__item` | Initiates form element vertical positioning. |
```
Comment (srambach) [inline src/patternfly/components/Toolbar/examples/Toolbar.md:660]: Stupid nit, but can you change "initiates" to "Modifies for" or whatever makes sense grammatically? (maybe the ones below too?)

**Review by srambach — APPROVED** [review]
Just the one nit about the documentation, which isn't absolutely necessary. Nice improvement ❇️

**Review by srambach — APPROVED** [review]
👍

**Pull request merged**